src/witan/models/dem/ccm/fert/hist_asfr_age.clj

```diff
@@ -9,14 +9,55 @@
             [incanter.core :as i]))
 
 (defn ->births-data-year
-  [{:keys [births-data]}]
   "Takes births-data dataset
-   Returns maximum value in Year column of births-data")
+   Returns maximum value in Year column of births-data"
+  [{:keys [births-data]}]
+  {:yr (->> births-data ;; FIXME: change key to :births-data-year once using workflow
+            (i/$ :year)
+            (reduce max))})
+
+(defn ->at-risk-this-year
+  "Filters ds for actualyear = yr, removes three columns
+  and rename two columns."
+  [{:keys [at-risk-popn yr]}]
+  {:at-risk-this-year (-> at-risk-popn
+                          (i/query-dataset {:actualyear yr})
+                          (ds/remove-columns [:age])
+                          (ds/rename-columns {:actualage :age :popn :popn-this-yr})
+                          (ds/remove-columns [:year :actualyear]))})
+
+(defn ->at-risk-last-year
+  "Filters ds for actualyear = yr - 1, removes three columns
+  and rename one column."
+  [{:keys [at-risk-popn yr]}]
+  {:at-risk-last-year (-> at-risk-popn
+                          (i/query-dataset {:actualyear (dec yr)})
+                          (ds/rename-columns {:popn :popn-last-yr})
+                          (ds/remove-columns [:actualage :actualyear]))})
+
+(defn ->births-pool
```

Comment:
great that you are using sensible names - realise the way the code is written may not be exactly how we would like to write fns in Clojure (though I think luckily for us its v close!)